calabash-cucumber/lib/calabash-cucumber/console_helpers.rb

```diff
@@ -155,6 +155,10 @@ def puts_console_details
       # @param {Integer} max_width
       def print_marks(marks, max_width)
         counter = -1
+        # Replace nil elements with empty string to avoid "ArgumentError: comparison of Array with Array failed" error
+        marks.each { |elm|
+          elm.map! {|e| e ? e : ''}
+        }
         marks.sort_by.each { |elm|
```

Comment:
Shouldn't you replace at this line `sort_by` with 'sort'?
I think this is the root cause for failing tests ;)